sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -78,6 +78,27 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     }
   }
 
+  private def toSQL(node: LogicalPlan, topNode: Boolean): String = {
```

Comment:
After rethinking this issue, I believe SQL generation of partial **optimized** plans is very important for federation to RDBMS. However, SQL generation of optimized plans is much more complex. So far, it is not the purpose. I am wondering if SQL generation of partial **analyzed** plans is needed?